### PR TITLE
added the ObjectNet benchmark

### PR DESCRIPTION
**Pull request by dmayo**

### CODE REVIEW

**Comment by dapello** [comment]
I pushed what seems to be a working version, including tests like those for ImageNet. Let's get this baby running!

File: brainscore/benchmarks/objectnet.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+class Objectnet(BenchmarkBase):
+    def __init__(self):
+        self._stimulus_set = brainscore.get_stimulus_set('katz.BarbuMayo2019')
+        self._similarity_metric = Accuracy()
+        ceiling = Score([1, np.nan], coords={'aggregation': ['center', 'error']}, dims=['aggregation'])
+       
+        super(Objectnet, self).__init__(identifier='katz.BarbuMayo2019-top1', version=1,
+                                           ceiling_func=lambda: ceiling,
+                                           parent='ObjectNet',
```
Comment (mschrimpf) [inline brainscore/benchmarks/objectnet.py:29]: parent should be `'engineering'`

File: brainscore/benchmarks/objectnet.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                                                    year      = {2019},
+                                                    url       = {https://proceedings.neurips.cc/paper/2019/hash/97af07a14cacba681feacf3012730892-Abstract.html},
+                                                    }""")
+
+
+    def __call__(self, candidate):
+        # The proper `fitting_stimuli` to pass to the candidate would be the imagenet training set.
+        # For now, since almost all models in our hands were trained with imagenet, we'll just short-cut this
+        # by telling the candidate to use its pre-trained imagenet weights.
+        candidate.start_task(BrainModel.Task.label, 'imagenet')
```
Comment (mschrimpf) [inline brainscore/benchmarks/objectnet.py:52]: is there perfect overlap between imagenet labels and objectnet labels? If not, should this be taken into account in the similarity metric / or even better when instructing the model about the task?

File: packaging/katz/BarbuMayo2019.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+image_paths = {}  # collect mapping of image_id to filepath
+stimuli_directory = "images_no_border"
+
+with open('mappings/objectnet.csv' ,'r') as f:
+    df = pd.read_csv(f)
+    df = df.reset_index(False)
+    df = df.rename(columns={"index":"image_id", "imagenet_label":"synset", "file_name":"filepath"})
+    df = df.astype({'image_id':'str'})
+    df['filepath'] = 'images_no_border/'+df['filepath']
+    #print(df)
```
Comment (mschrimpf) [inline packaging/katz/BarbuMayo2019.py:17]: dead code, but it's in packaging, so I'm not going to chase you

**Review by mschrimpf — APPROVED** [review]
small change request + fix the unit test and this should be good to go

File: brainscore/benchmarks/objectnet.py (hunk 2 above)
Comment (dapello) [reply brainscore/benchmarks/objectnet.py:52]: the synset labels for the objectnet categories we used are perfectly aligned with the ImageNet ones. is that what you're asking?

**Comment by dmayo** [comment]
There is not a perfect overlap in the labels. Sometimes imagenet labels are either a bit more specific or they are a list of things that aren't a perfect fit for objectnet labels. I don't have a metric for how good a mapping is. I do know that the imagenet class that an objectnet class is mapped to is the most frequent imagent class prediction for that objectnet class. This means that at least the mapped imagent class is the best one of the 1k options, even if it's not quite perfect. Also many imagnet labels don't perfectly align with the actual images in the imagnet training set anyway. (Look at sunglass).

We could try an exact label match vs less exact label match objectnet metric, but I don't think we need it right now for brainscore.


-------- Original message --------
From: Martin Schrimpf ***@***.***>
Date: 4/6/22 2:41 PM (GMT-05:00)
To: brain-score/brain-score ***@***.***>
Cc: David Mayo ***@***.***>, Author ***@***.***>
Subject: Re: [brain-score/brain-score] added the ObjectNet benchmark (PR #318)


@mschrimpf approved this pull request.

small change request + fix the unit test and this should be good to go

________________________________

In brainscore/benchmarks/objectnet.py<https://github.com/brain-score/brain-score/pull/318#discussion_r844269251>:

> +from brainio.lookup import lookup_stimulus_set
+
+NUMBER_OF_TRIALS = 10
+
+_logger = logging.getLogger(__name__)
+LOCAL_STIMULUS_DIRECTORY = '/braintree/data2/active/common/objectnet-stimuli/'
+
+class Objectnet(BenchmarkBase):
+    def __init__(self):
+        self._stimulus_set = brainscore.get_stimulus_set('katz.BarbuMayo2019')
+        self._similarity_metric = Accuracy()
+        ceiling = Score([1, np.nan], coords={'aggregation': ['center', 'error']}, dims=['aggregation'])
+
+        super(Objectnet, self).__init__(identifier='katz.BarbuMayo2019-top1', version=1,
+                                           ceiling_func=lambda: ceiling,
+                                           parent='ObjectNet',


parent should be 'engineering'

________________________________

In brainscore/benchmarks/objectnet.py<https://github.com/brain-score/brain-score/pull/318#discussion_r844270512>:

> +                                                                Josh Tenenbaum and
+                                                                Boris Katz},
+                                                    title     = {ObjectNet: {A} large-scale bias-controlled dataset for pushing the
+                                                                limits of object recognition models},
+                                                    booktitle = {NeurIPS 2019},
+                                                    pages     = {9448--9458},
+                                                    year      = {2019},
+                                                    url       = {https://proceedings.neurips.cc/paper/2019/hash/97af07a14cacba681feacf3012730892-Abstract.html},
+                                                    }""")
+
+
+    def __call__(self, candidate):
+        # The proper `fitting_stimuli` to pass to the candidate would be the imagenet training set.
+        # For now, since almost all models in our hands were trained with imagenet, we'll just short-cut this
+        # by telling the candidate to use its pre-trained imagenet weights.
+        candidate.start_task(BrainModel.Task.label, 'imagenet')


is there perfect overlap between imagenet labels and objectnet labels? If not, should this be taken into account in the similarity metric / or even better when instructing the model about the task?

________________________________

In packaging/katz/BarbuMayo2019.py<https://github.com/brain-score/brain-score/pull/318#discussion_r844270901>:

> +from brainio.stimuli import StimulusSet
+from brainio.packaging import package_stimulus_set
+
+import csv
+import pandas as pd
+stimuli = []  # collect meta
+image_paths = {}  # collect mapping of image_id to filepath
+stimuli_directory = "images_no_border"
+
+with open('mappings/objectnet.csv' ,'r') as f:
+    df = pd.read_csv(f)
+    df = df.reset_index(False)
+    df = df.rename(columns={"index":"image_id", "imagenet_label":"synset", "file_name":"filepath"})
+    df = df.astype({'image_id':'str'})
+    df['filepath'] = 'images_no_border/'+df['filepath']
+    #print(df)


dead code, but it's in packaging, so I'm not going to chase you

-

**Comment by mschrimpf** [comment]
not sure what happened with travis here, re-triggering

File: tests/test_benchmarks/test_objectnet.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import numpy as np
+import pytest
+
+from brainio.assemblies import BehavioralAssembly
+from brainscore.benchmarks.objectnet import Objectnet
+from brainscore.model_interface import BrainModel
+
+@pytest.mark.private_acces
```
Comment (mschrimpf) [inline tests/test_benchmarks/test_objectnet.py:8]: @dapello typo, this should be `access` (two s) at the end. I think this is the last thing then all is good
